cmd/nonmem.go

```diff
@@ -137,122 +133,120 @@ var parallelRegexesToRemove []string = []string{
+	// Model is the name of the model on which we will action: acop.mod
 	Model string `json:"model_name"`
-	//OriginalModel is the original filename, primarily used for NMQual or others that require changing filenames
+	// OriginalModel is the original filename, primarily used for NMQual or others that require changing filenames
 	OriginalModel string `json:"original_model"`
-	//Path is the Fully Qualified Path to the original model
+	// Path is the Fully Qualified Path to the original model
 	Path string `json:"model_path"`
 	// DataPath is the path to the data when executing the model
 	DataPath string `json:"data_path"`
 	// DataMD5 is the md5hash of the data
 	DataMD5 string `json:"data_md5"`
-	//ModelMD5 is the digest of the executing model file
+	// ModelMD5 is the digest of the executing model file
 	ModelMD5 string `json:"model_md5"`
-	//FileName is the Filename component (sans extension)
+	// FileName is the Filename component (sans extension)
 	FileName string `json:"model_filename"`
-	//Extension is the extension of the file
+	// Extension is the extension of the file
 	Extension string `json:"model_extension"`
-	//OriginalPath is the path at which the original model was located: /Users/Documents/acop/
+	// OriginalPath is the path at which the original model was located: /Users/Documents/acop/
 	OriginalPath string `json:"original_path"`
-	//OutputDir is the directory into which the copied models and work will be located
+	// OutputDir is the directory into which the copied models and work will be located
 	OutputDir string `json:"output_dir"`
-	//Settings are basically the cobra definitions / requirements for the iteration
+	// Settings are basically the cobra definitions / requirements for the iteration
 	Configuration configlib.Config `json:"configuration"`
-	//Whether or not the model had an error on generation or execution
+	// Whether or not the model had an error on generation or execution
 	Error error `json:"error"`
 }
 
 var nonmemLongDescription string = fmt.Sprintf("\n%s\n\n%s\n\n%s\n", runLongDescription, summaryLongDescription, covcorLongDescription)
 
-// RunCmd represents the run command
+// RunCmd represents the run command.
 var nonmemCmd = &cobra.Command{
 	Use:   "nonmem",
 	Short: "nonmem a (set of) models locally or on the grid",
 	Long:  nonmemLongDescription,
 	Run:   nonmem,
 }
 
-func nonmem(cmd *cobra.Command, args []string) {
+func nonmem(_ *cobra.Command, _ []string) {
 	println(nonmemLongDescription)
 }
 
 func init() {
 	RootCmd.AddCommand(nonmemCmd)
 
-	//NM Selector
+	// NM Selector
 	const nmVersionIdentifier string = "nm_version"
 	nonmemCmd.PersistentFlags().String(nmVersionIdentifier, "", "Version of nonmem from the configuration list to use")
-	viper.BindPFlag(nmVersionIdentifier, nonmemCmd.PersistentFlags().Lookup(nmVersionIdentifier))
+	errpanic(viper.BindPFlag(nmVersionIdentifier, nonmemCmd.PersistentFlags().Lookup(nmVersionIdentifier)))
```

Comment:
I really had to think through the use of this in the execution goroutines a lot. For local execution it would seem to be fine because you want to interrupt everything if something fails, but in SGE execution I had to step back and think about it again. The initial phase that the user interacts with just basically puts a script onto the SGE nodes that runs locally there, so if failures occur there, it's all written out into the out file. So even if *something* failed and we panic'd in the goroutines for execution, it would still only affect that run. 

What it would do is interfere with the post execution scripts, which I don't know if is really a problem since I don't know of anyone using them. The panic would basically keep the orchestrator from getting to the point of running the script and indicating a status of failure

Reply:
Well if a panic is bad, what about a silent failure?

Reply:
> A silent failure would still be fine. What would happen is any error state would be wrapped up in the status of the job by the manager and collected into the post execution script

I think this may be rendered moot as all of these panics would happen on create of the rootcmd and its dependent children. In other words, the application will **always** panic on startup or **never** panic on startup.